templates/include/glean/glean.1.schema.json

```diff
@@ -54,11 +54,17 @@
         "end_time": @GLEAN_DATETIME_1_JSON@,
         "experiments": {
           "type": "object",
-          "propertyNames": @GLEAN_SHORT_ID_1_JSON@,
+          "propertyNames": {
```

Comment:
This also suggests we were being too permissive in the Glean SDK implementation, as we were not erroring out on these. Right? The change sounds good to me, unless it has deep implications for pipeline. @fbertsch will tell.

Reply:
That's correct -- we don't to the regex check at runtime.  But it would be hard to raise an error because, like most Glean SDK calls, it's asynchronous and returns nothing.  The best we could probably do is drop the experiment from the `experiments` object.

@fbertsch would know for sure, but I doubt it has deep implications for the pipeline -- the Presto ingestion is *already* behaving this way.  It would only change the behavior on BQ.

Reply:
I assume this ends up as a pseudo map type in BQ, in which case the experiment name ends up in a string field and can be anything. So I don't see any technical problem with the pipeline by relaxing this.

But there _is_ a data curation cost to having different experiments use different conventions. There will likely be bugs introduced in analyses where we assume one convention is being used and miss some experiments data because of it. It would be great to be able to enforce a strong convention at the SDK level, but assuming that's not possible, this may be a fight we simply have to let ourselves lose.

Reply:
I'm open to discussing whether this *should* restrict things and what those restrictions should be.  In part this perhaps my overreaction to realizing we lost 2/3 of our data for nearly a month :)

With [rejection monitoring](https://bugzilla.mozilla.org/show_bug.cgi?id=1581554) in place, client-side validation (if possible), and validation of the data that goes into an experiment, we could more safely restrict this.

I'm going to mark this as WIP for now while we figure all that out...

Reply:
I don't know what previous conventions we have in place for experiment names, so I don't have an opinion as to what the convention here should be, just that it will avoid some problems down the line if experiment names are all consistent.

That said, I would agree that losing wide swaths of pings is a bigger problem than ensuring we have clean experiment names, so I totally understand if being more permissive in validation here seems like the most pragmatic option.

Reply:
Removing this is only removing the regex, so we're good to go on that end.

We need to have a broader discussion on what we're trying to do when we put field limitations in a schema here. are we:
1. Trying to catch pathological clients
2. Trying to prevent non schema-compatible pings from entering the pipeline downstream
3. Trying to make analysis easier by limiting field sets

IMO we are mostly trying to do (1.) and (2.). Doing (3.) is perhaps a job for downstream datasets, however I'm open to alternatives.

tl;dr: this is fine to deploy